core/src/main/java/org/jruby/javasupport/binding/Initializer.java

```diff
@@ -489,7 +489,9 @@ private static void setJavaClassFor(final Class<?> javaClass, final RubyModule p
         for ( Class<?> klass = javaClass; klass != null; klass = klass.getSuperclass() ) {
             // only add class's methods if it's public or we can set accessible
             // (see JRUBY-4799)
-            if (Modifier.isPublic(klass.getModifiers()) || JavaUtil.CAN_SET_ACCESSIBLE) {
+            if (Modifier.isPublic(klass.getModifiers())
+                    || JavaUtil.CAN_SET_ACCESSIBLE
+                    || Modules.getModule(klass).isOpen(klass.getPackage().getName(), Modules.getModule(Ruby.class))) {
```

Comment:
will NPE on Java 8 -> `getPackage()` returns null for default package while 9 returns a package object
recall, we have logic dealing with this in `com.headius.backport9.modules` itself, so maybe having a 
`Modules.isOpen(Class target, Class from)` there handling the details would fit well ...

Reply:
Good call, I'll force push that change.